test/libcommonserver/io/testchecksetgetrights.cpp

```diff
@@ -0,0 +1,267 @@
+/*
+ * Infomaniak kDrive - Desktop
+ * Copyright (C) 2023-2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "testio.h"
+#include "libcommonserver/utility/utility.h"
+
+#include <filesystem>
+
+using namespace CppUnit;
+
+namespace KDC {
+
+static struct RightsSet {
+        RightsSet(int rights) : read(rights & 4), write(rights & 2), execute(rights & 1){};
+        RightsSet(bool read, bool write, bool execute) : read(read), write(write), execute(execute){};
+        bool read;
+        bool write;
+        bool execute;
+};
+
+void TestIo::testCheckSetAndGetRights() {
+#ifdef _WIN32
+    CPPUNIT_ASSERT(Utility::init());  // Initialize the utility library, needed to access/change the permissions on Windows
+#endif
+    const TemporaryDirectory temporaryDirectory;
+
+    // Test if the rights are correctly set and get on a directory
+    {
+        const SyncPath path = temporaryDirectory.path / "changePerm";
+
+        IoError ioError = IoErrorUnknown;
+        CPPUNIT_ASSERT(IoHelper::createDirectory(path, ioError));
+        CPPUNIT_ASSERT_EQUAL(IoErrorSuccess, ioError);
+
+        bool isReadable = false;
+        bool isWritable = false;
+        bool isExecutable = false;
+        bool exists = false;
+
+        RightsSet rightsSet(false, false, false);
+
+        // For a directory
+
+        /* Test all the possible rights and all the possible order of rights modification. ie:
+         *  | READ | WRITE | EXECUTE | |
+         *  |  0   |   0   |    0    | v
+         *  |  0   |   0   |    1    | |
+         *  |  0   |   0   |    0    | v
+         *  |  0   |   1   |    0    | |
+         *  |  0   |   0   |    0    | v
+         *  |  0   |   1   |    1    | |
+         *  |  0   |   0   |    0    | v
+         *  |  1   |   0   |    0    | |
+         *  |  0   |   0   |    0    | v
+         *  |  1   |   0   |    1    | |
+         *  |  0   |   0   |    0    | v
+         *  |  1   |   1   |    0    | |
+         *  |  0   |   0   |    0    | v
+         *  |  1   |   1   |    1    | |
+         *  |  0   |   0   |    0    | v
+         *  ...
+         */
+
+        for (int baseRigths = 0; baseRigths < 7; baseRigths++) {
+            for (int targetRigths = baseRigths + 1; targetRigths < 8; targetRigths++) {
+                rightsSet = RightsSet(baseRigths);
```

Comment:
```suggestion
                auto rightsSet = RightsSet(baseRigths);
```
Then you can remove the declaration above.  It's usually safer and simpler to declare variables where they are actually used.

Reply:
Thanks you for the suggestion. (https://github.com/Infomaniak/desktop-kDrive/pull/68/commits/99cfbb61674dc6c137fcc411837895b67842c5ad)